packages/botonic-plugin-contentful/src/nlp/normalizer.ts

```diff
@@ -90,17 +93,94 @@ export class NormalizedUtterance {
   }
 }
 
+export abstract class Preprocessor {
+  abstract preprocess(txt: string): string
+}
+
+export class NopPreprocessor {
+  preprocess(txt: string): string {
+    return txt
+  }
+}
+
+/**
+ * Removes dots within acronyms, even if missing last dot,
+ * or immediately followed by a different separator
+ */
+export class AcronymPreprocessor implements Preprocessor {
+  private static readonly DOT = '.'
+  private SEPS_NO_DOT: string
+  constructor(separators: string) {
+    this.SEPS_NO_DOT = separators.replace(AcronymPreprocessor.DOT, '')
+  }
+
+  preprocess(txt: string): string {
+    if (!txt.includes(AcronymPreprocessor.DOT)) return txt
+    const wordsAndSeparators = this.splitWordsAndSeparators(txt)
+    txt = ''
+    for (const wOrSep of wordsAndSeparators) {
+      const isSeparator = wOrSep.includes(this.SEPS_NO_DOT)
+      if (!isSeparator) {
+        txt = txt + this.preprocessWord(wOrSep)
+      } else {
+        txt = txt + wOrSep
+      }
+    }
+    return txt
+  }
+
+  private splitWordsAndSeparators(txt: string): string[] {
```

Comment:
The same way we have the `DEFAULT_SEPARATORS_REGEX` constant, maybe it would be better to create a `RegExp` with the separators without the dot and use a `split` instead of using this function? The result should be the same, right?

Reply:
We cannot use split because this function does not "by the separators", it splits "the words AND the separators"
eg. splitWordsAndSeparators("aa.bb") returns ["aa", ".", "bb"]
Not loosing the separator makes easier the reconstruction of the full input in AcronymPreprocessor.preprocess

Reply:
You're totally right!